### DISCUSSION TITLE
Can I get test results as json

### DISCUSSION THREAD

**Discussion by benjamin-inspectly** [leancodepl/patrol#1598] [Q&A] [answered]
Hi!

I'm trying to get Patrol to work on Codemagic CI and it is surprisingly easy.
However Codemagic has a test report feature built in, if just I can get the testresults as a json-file instead of the default html file.

It is possible with flutter test - but is it possible with patrol somehow? I checked the "patrol help" to see if I missed an argument in the "patrol test" command.

Documentation from Codemagic: https://docs.codemagic.io/yaml-testing/testing/

**Comment by fylyppo** [comment]
Hi, speaking of Codemagic, you can provide xml files as a test report.
From patrol as a runner you'll get only html and xml or xcresult, so there's no support for json.

**Comment by benjamin-inspectly** [comment]
Hi @fylyppo  :) Thanks for answering!
How do I get it to return an xml?

In this folder, I only get an html? :)
build\app\reports\androidTests\connected\flavors\appCenter

**Reply by fylyppo** [reply, marked as answer]
iOS .xcresult is placed at `~/Library/Developer/Xcode/DerivedData/**/Logs/Test/*.xcresult`
Android JUnit .xml is placed in your Flutter project `/build/app/outputs/androidTest-results/connected/*.xml`

**Reply by benjamin-inspectly** [reply]
Thank you so much for your reply! :) works perfectly with codemagic :)
